src/app/features/import/import-column-mapping/import-column-mapping.component.spec.ts

```diff
@@ -0,0 +1,87 @@
+import { ComponentFixture, TestBed } from "@angular/core/testing";
+
+import { ImportColumnMappingComponent } from "./import-column-mapping.component";
+import { MockedTestingModule } from "../../../utils/mocked-testing.module";
+import { ImportModule } from "../import.module";
+import { MatDialog } from "@angular/material/dialog";
+import { EnumValueMappingComponent } from "./enum-value-mapping/enum-value-mapping.component";
+import { Child } from "../../../child-dev-project/children/model/child";
+import { SimpleChange } from "@angular/core";
+import { ColumnMapping } from "../column-mapping";
+
+describe("ImportMapColumnsComponent", () => {
+  let component: ImportColumnMappingComponent;
+  let fixture: ComponentFixture<ImportColumnMappingComponent>;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [MockedTestingModule, ImportModule],
+      declarations: [ImportColumnMappingComponent],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(ImportColumnMappingComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+
+    spyOn(component.columnMappingChange, "emit");
+  });
+
+  it("should reset invalid mappings when new entityType selected", () => {
+    component.columnMapping = [
+      { column: "x", propertyName: "name" }, // property also exists on new entityType
+      { column: "y", propertyName: "projectNumber" }, // property does not exist on new entityType
+    ];
+
+    component.entityType = "School";
+    component.ngOnChanges({
+      entityType: new SimpleChange("Child", "School", false),
+    });
+
+    const expectedNewMapping: ColumnMapping[] = [
+      { column: "x", propertyName: "name" },
```

Comment:
the problem is that the mapping assumes a very specific input rawData (the imported column names). Applying a previous mapping will only work if the new imported spreadsheet has 100% same columns as the historic one. Otherwise new columns will be inaccessible to re-map. And I am not sure what would happen to mappings that do not have an actual column anymore.

We could of course also just disallow to load a previous mapping for any case where imported columns do not match exactly.

Reply:
yes, you are probably right. I am not really sure anymore why I thought this was needed